spmd/compiler/distribute.py

```diff
@@ -307,7 +308,34 @@ def _rebuild_graph(
                     assert (
                         len(dtn.args) == 1
                     ), f"Expecting single output, but got {dtn.args} {len(dtn.args[0])}"
-                    node.replace_all_uses_with(value_remap[dtn.args[0][0]])
+                    outputs = dtn.args[0]
+                    # we currently support two very specific types of output
+                    # 1. single output
+                    # 2. multiple outputs resulting from getitem of all elements of tuple
+                    if len(outputs) == 1:
+                        # for single output, we replace the node with the single node
+                        output = outputs[0]
+                    else:
+                        # for multiple outputs, we check that these outputs correspond
+                        # to all elements of a tuple. In that case, we replace
+                        # uses of the output directly with the original tuple
+                        source = None
+                        for i, o in enumerate(outputs):
```

Comment:
Very minor nit: I think 'out' instead of 'o' is more informative and avoids o/0/O misreading, esp since there's o.args[0] multiple times.